SQL/New_patches/2018-09-07_EEG_tables.sql

```diff
@@ -0,0 +1,259 @@
+  `PhysiologicalStatusTypeID`  TINYINT(2) UNSIGNED DEFAULT NULL,
+  `Name`                       VARCHAR(50)         NOT NULL,
+  `Description`                VARCHAR(255)        DEFAULT NULL,
+  `SamplingFrequency`          INT(6)              DEFAULT NULL,
+  `LowCutoff`                  DECIMAL(8,3)        DEFAULT NULL,
+  `HighCutoff`                 DECIMAL(8,3)        DEFAULT NULL,
+  `ManualFlag`                 DECIMAL(5,4)        DEFAULT NULL,
+  `Notch`                      INT(6)              DEFAULT NULL,
+  `StatusDescription`          VARCHAR(255)        DEFAULT NULL,
+  `Unit`                       VARCHAR(255)        DEFAULT NULL,
+  `File`                       VARCHAR(255)        DEFAULT NULL,
+  `SoftwareFilters`            VARCHAR(255)        DEFAULT NULL,
+  PRIMARY KEY (`PhysiologicalChannelID`),
+  CONSTRAINT `FK_physiological_file_PhysiologicalFileID_2`
+    FOREIGN KEY (`PhysiologicalFileID`)
+    REFERENCES `physiological_file` (`PhysiologicalFileID`)
+    ON DELETE CASCADE,
+  CONSTRAINT `FK_physiological_channel_type_PhysiologicalChannelTypeID`
+    FOREIGN KEY (`PhysiologicalChannelTypeID`)
+    REFERENCES `physiological_channel_type` (`PhysiologicalChannelTypeID`),
+  CONSTRAINT `FK_physiological_status_type_PhysiologicalStatusTypeID`
+    FOREIGN KEY (`PhysiologicalStatusTypeID`)
+    REFERENCES `physiological_status_type` (`PhysiologicalStatusTypeID`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+
+
+-- Create a physiological_electrode table
+CREATE TABLE `physiological_electrode` (
+  `PhysiologicalElectrodeID` INT(10) UNSIGNED NOT NULL      AUTO_INCREMENT,
+  `PhysiologicalFileID`      INT(10) UNSIGNED NOT NULL,
+  `Name`                     VARCHAR(50)      NOT NULL,
+  `X`                        DECIMAL(12,6)    NOT NULL,
+  `Y`                        DECIMAL(12,6)    NOT NULL,
+  `Z`                        DECIMAL(12,6)    NOT NULL,
+  `Type`                     VARCHAR(50)      NOT NULL,
+  `Material`                 VARCHAR(50)      NOT NULL,
+  `File`                     VARCHAR(255)     DEFAULT NULL,
+  PRIMARY KEY (`PhysiologicalElectrodeID`),
+  CONSTRAINT `FK_physiological_file_PhysiologicalFileID_3`
+    FOREIGN KEY (`PhysiologicalFileID`)
+    REFERENCES `physiological_file` (`PhysiologicalFileID`)
+    ON DELETE CASCADE
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+
+
+-- Create physiological_task_event table that will store all information
+-- regarding the task executed during the physiological recording
+CREATE TABLE `physiological_task_event` (
+  `PhysiologicalTaskEventID` INT(10) UNSIGNED NOT NULL       AUTO_INCREMENT,
+  `PhysiologicalFileID`      INT(10) UNSIGNED NOT NULL,
+  `Onset`                    DECIMAL(11,6)    NOT NULL,
+  `Duration`                 DECIMAL(11,6)    NOT NULL,
+  `TrialType`                VARCHAR(255)     DEFAULT NULL,
+  `ResponseTime`             TIME             DEFAULT NULL,
+  `EventCode`                VARCHAR(255)     DEFAULT NULL,
```

Comment:
a 255 char length for code? I need more details

Reply:
oups. Looks like it is an integer. Not sure how big it can be though. Setting it to 10 for now